packages/calling-component-bindings/src/incomingCallStackSelector.ts

```diff
@@ -0,0 +1,67 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import { CallClientState, IncomingCallState } from '@internal/calling-stateful-client';
+/* @conditional-compile-remove(one-to-n-calling) */
+import { TeamsIncomingCallState } from '@internal/calling-stateful-client';
+import { getIncomingCalls, getRemovedIncomingCalls } from './baseSelectors';
+import { createSelector } from 'reselect';
+import { ActiveIncomingCall } from '@internal/react-components';
+
+/**
+ * Selector to get the active and removed incoming calls.
+ * @beta
+ */
+export type IncomingCallStackSelector = (state: CallClientState) => {
+  activeIncomingCalls: ActiveIncomingCall[];
+  removedIncomingCalls: ActiveIncomingCall[];
+};
+
+/**
+ * Select the active and removed incoming calls from the stateful client for the IncomingCallNotificationStackComponent.
+ * @beta
+ */
+export const incomingCallStackSelector: IncomingCallStackSelector = createSelector(
+  [getIncomingCalls, getRemovedIncomingCalls],
+  (
+    incomingCalls,
+    removedIncomingCalls
+  ): {
+    activeIncomingCalls: ActiveIncomingCall[];
+    removedIncomingCalls: ActiveIncomingCall[];
+  } => {
+    // Convert incoming call state to active incoming call
+    const componentIncomingCalls = incomingCalls.map(
+      (
+        incomingCall: IncomingCallState | /* @conditional-compile-remove(one-to-n-calling) */ TeamsIncomingCallState
+      ) => {
+        return {
+          id: incomingCall.id,
+          callerInfo: {
+            displayName: incomingCall.callerInfo.displayName || 'Unknown Caller'
+          },
+          startTime: incomingCall.startTime,
+          endTime: incomingCall.endTime
+        };
+      }
+    );
+    const componentRemovedIncomingCalls = removedIncomingCalls.map(
+      (
+        incomingCall: IncomingCallState | /* @conditional-compile-remove(one-to-n-calling) */ TeamsIncomingCallState
+      ) => {
+        return {
+          id: incomingCall.id,
+          callerInfo: {
+            displayName: incomingCall.callerInfo.displayName || 'Unknown Caller'
```

Comment:
Wondering why we don't do this process in stateful client layer

Reply:
The content that we are storing in state for the stateful client should reflect the SDK type as state no? this is here so that we are returning only the information that the Component needs